### PR TITLE
feat(minifier): remove unnecessary "use strict"

### PR DESCRIPTION
**Pull request by sapphi-red**

### CODE REVIEW

**Comment by sapphi-red** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8173?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8174** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8174?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8173** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8173?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8173?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8172** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8172?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8173).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/oxc_minifier/src/ast_passes/peephole_remove_dead_code.rs [hunk 1]
```diff
@@ -124,6 +128,16 @@ impl<'a, 'b> PeepholeRemoveDeadCode {
...
 
+    /// Drop `"use strict";` directives if the function is empty.
+    fn drop_use_strict_directives_if_function_is_empty(
+        body: &mut FunctionBody<'a>,
+        _ctx: &mut TraverseCtx<'a>,
+    ) {
+        if body.statements.is_empty() {
+            body.directives.retain(|directive| !directive.is_use_strict());
+        }
+    }
```
Comment (sapphi-red) [inline crates/oxc_minifier/src/ast_passes/peephole_remove_dead_code.rs:139]: I'm not sure if it makes sense to exist in PeepholeRemoveDeadCode. This has to run after everything that might remove statements.

**Comment by Boshen** [comment]
This is not ideal, I'll try something else later.

**Comment by sapphi-red** [comment]
@Boshen Did you try this one?

**Comment by Boshen** [comment]
> @Boshen Did you try this one?

nope

**Pull request closed without merging**